cmd/otel-allocator/allocation/least_weighted.go

```diff
@@ -94,13 +98,13 @@ func (allocator *leastWeightedAllocator) findNextCollector() *Collector {
 // This method is called from within SetTargets and SetCollectors, which acquire the needed lock.
 // This is only called after the collectors are cleared or when a new target has been found in the tempTargetMap.
 // INVARIANT: allocator.collectors must have at least 1 collector set.
-func (allocator *leastWeightedAllocator) addTargetToTargetItems(target *TargetItem) {
+func (allocator *leastWeightedAllocator) addTargetToTargetItems(tg *target.Item) {
```

Comment:
It might be good to separate out the refactoring of `TargetItem` into `target.Item` into a new PR so that it can be assessed independently of the changes required to add target filtering.

Reply:
Yeah thats a good point that the renaming and moving of objects can be confusing to reviewers.  The target package is now necessary to resolve an import cycle between allocation and prehook packages.  So now I changed things to `target.TargetItem` and `target.NewTargetItem` to hopefully minimize any confusion.  Let me know if you think there's a better way to go about this.

Reply:
I think that's fine, but I'd like to see a separate PR be merged first that only moves these types into the new `target` package.  That will separate out the refactoring from the new functionality and hopefully make it easier to review the new functionality without noise from the refactoring.

Reply:
Sounds good @Aneurysm9 I created this [issue](https://github.com/open-telemetry/opentelemetry-operator/issues/1213) to track this change, and will create a new PR with the new `target` package. Thanks for the feedback!